api/python/provisioner/commands/upgrade/get_swupgrade_info.py

```diff
@@ -0,0 +1,142 @@
+class GetSWUpgradeInfo(CommandParserFillerMixin):
+    """
+    Base class that provides API for getting SW upgrade repository information.
+
+    """
+
+    input_type: Type[inputs.NoParams] = inputs.NoParams
+    _run_args_type = GetSWUpgradeInfoArgs
+
+    @staticmethod
+    def _get_package_version(release: str) -> dict:
+        """
+        Function returns information about CORTX packages and their versions.
+
+        Parameters
+        ----------
+        release : str
+            SW upgrade repository version
+
+        Returns
+        -------
+        dict
+            return dictionary with CORTX packages and their versions
+
+        """
+        repo = f"sw_upgrade_{CORTX_ISO_DIR}_{release}"
+
+        cmd = (f"yum repo-pkgs {repo} list 2>/dev/null | "
+               f"grep '{repo}' | awk '{{print $1\" \"$2}}'")
+
+        res = cmd_run(cmd, targets=local_minion_id(),
+                      fun_kwargs=dict(python_shell=True))
+
+        packages = res[local_minion_id()].strip().split('\n')
+
+        if packages:
+            logger.debug(f"List of packages in repository '{repo}':"
+                         f" {packages}")
+        else:
+            logger.debug(f"There are no packages in repository '{repo}'")
+
+        res = dict()
+        # NOTE: Format is following
+        # ```
+        #  {
+        #      'cortx-motr': {
+        #             'version': '2.0.0-277',
+        #          },
+        #  }
+        # ```
+        #
+        # TODO: Along the with 'version', field we need to add
+        #  'constraint version' field to provide necessary information about
+        #  compatibility with old versions
+        for entry in packages:
+            pkg, ver = entry.split(" ")
+            res[pkg] = {SWUpgradeInfoFields.VERSION.value: ver}
+
+        return res
+
+    def run(self, release: str = None) -> dict:
+        """
+        Main function for Get SW Upgrade Repo command. Command returns
+        information about CORTX packages and their versions.
+
+        Parameters
+        ----------
+        release: str
+            SW upgrade repository release version
+
+        Returns
+        -------
+        dict:
+            return dictionary with CORTX packages and their versions
+
+        """
+        local_minion = local_minion_id()
+
+        if release is None:
+            # NOTE: take the latest release from SW upgrade repositories
+
+            # TODO: make get pillar API public and available for others to
+            #  avoid code duplication
+            pillars = SWUpgrade._get_pillar('release/upgrade', local_minion)
+
+            upgrade_releases = pillars[local_minion].get("repos", None)
+
+            if not upgrade_releases:
+                raise ValueError("release/upgrade/repos pillars are missed")
+
+            upgrade_releases = list(upgrade_releases.keys())
+            upgrade_releases.remove(REPO_CANDIDATE_NAME)
+
+            release = max(upgrade_releases, key=version.parse)
```

Comment:
@dmitrydb Cool! But would work only in case Cortx release versions satisfy PEP-440 which is likely but who knows... Could you make a note here? Thanks

Reply:
I added a note here about our assumption regarding release version formatting. But I have some doubts that the current format is valid according to PEP-440. Fortunately `packaging` parses it correctly:
```python
from packaging import version

a = version.parse('1.1.1-207')
print(a)
print(a.post)
```
And output
```
1.1.1.post207
207
```